### DISCUSSION TITLE
[ZK Stack] Gas estimation issue

### DISCUSSION THREAD

**Discussion by ktarun1419** [zkSync-Community-Hub/zksync-developers#541] [ZK Stack]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Hi i am trying to run  test transfer function.  
```
contract TestMWC {
    function testTransfer(
        address to
    ) external payable {
        require(msg.value > 0, "fee required");

        (bool success, ) = payable(to).call{value: msg.value}("");
        require(success, "fee failed.");
    }
}
```
 and this error is coming 
```
 Error: processing response error (body="{\"jsonrpc\":\"2.0\",\"error\":{\"code\":3,\"message\":\"failed to validate the transaction. reason: Validation revert: Account validation error: Not enough gas for transaction validation```
 it only runs when i give 400000 gasLimit which is huge for a transfer function ,FYI i have 0.02 ETH balance


### Repo Link (Optional)

_No response_

### Additional Details

_No response_

**Comment by ktarun1419** [comment]
Hi @bxpana , Could you please check this issue

**Reply by bxpana** [reply]
Thanks will take a look and get back to you on this. 

Regarding the gas fee and refund you can read more about how that works in our docs https://docs.zksync.io/zk-stack/concepts/fee-mechanism.html

**Reply by bxpana** [reply]
Are you still having an issue with this?

**Comment by ktarun1419** [comment]
Also why zksync takes extra gas and refund it afterwards

**Reply by bxpana** [reply]
You can also learn more about how the refunds work on ZKsync here: https://docs.zksync.io/zk-stack/concepts/fee-mechanism#refunds-1
